### ISSUE TITLE
Fix command mangles file when jinja comment is present

### ISSUE THREAD

**Issue by atgmello** [sqlfluff/sqlfluff#854]
## Expected Behaviour
Correctly fix trailing commas from input file.

Input file `stg_payments.sql`:
```sql
with source as (

    {#-
    Normally we would select from the table here, but we are using seeds to load
    our data in this project
    #}
    select * from {{ ref('raw_payments') }}

),

renamed as (

    select
        id as payment_id
        , order_id
        , payment_method
        --`amount` is currently stored in cents, so we convert it to dollars
        , amount / 100 as amount

    from source

)

select * from renamed
```

`sqlfluff fix` command line output:
```bash
==== finding fixable violations ====
== [models/staging/stg_payments.sql] FAIL
L:  10 | P:   9 | L019 | Found leading comma. Expected only trailing.
L:  11 | P:   9 | L019 | Found leading comma. Expected only trailing.
L:  13 | P:   9 | L019 | Found leading comma. Expected only trailing.
==== fixing violations ====
3 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
== [models/staging/stg_payments.sql] PASS
Done. Please check your files to confirm.
```

Expected `stg_payments.sql` output:
```sql
with source as (

    {#-
    Normally we would select from the table here, but we are using seeds to load
    our data in this project
    #}
    select * from {{ ref('raw_payments') }}

),

renamed as (

    select
        id as payment_id,
        order_id,
        payment_method,
        --`amount` is currently stored in cents, so we convert it to dollars
        amount / 100 as amount

    from source

)

select * from renamed
```

## Observed Behaviour
Actual output:

```sql
with source as (

    {#-
    Normally we would select from the table here, but we are using seeds to load
    our data in this project
    #}
    select * from {{ ref('raw_payments'renamed as (

    select
        id as payment_id,
        order_id,
        payment_method,
        --`amount` is currently stored in cents, so we convert it to dollars
        amount / 100 as amount

    from source

) source

)

select * from renamed
```

## Steps to Reproduce
`sqlfluff fix stg_payments.sql`

## Version
qlffluff: 0.4.1
dbt: 0.19.0
python: 3.7.10

## Configuration
Tested with both
```
[sqlfluff]
templater = jinja
# dbt templating does not keep trailing new lines (L009)
exclude_rules = L009
```
and
```
[sqlfluff]
templater = dbt
# dbt templating does not keep trailing new lines (L009)
exclude_rules = L009
```

## Notes

When the comments from lines 3-6 are remove, `sqlfluff` works as expected.

**Comment by atgmello** [comment]
A few minutes after posting this and playing with the above SQL a bit more, I was able to pinpointed the real issue. `sqlfluff` actually has no problems with Jinja comments of the form `{# ... #}`.

It does, however, get lost with the pattern `{#- ... #}` from above. Removing the dash at the beginning of the comment stops `sqlfluff` from breaking.
This SQL file is straight from dbt's [Jaffle Shop](https://github.com/fishtown-analytics/jaffle_shop) tutorial. I wouldn't have guessed that their style of comment was the reason behind the breakage.

Not sure if a real issue that should be further investigated in the code or just a pattern that should be avoided when writing comments. Feel free to close this issue if needed.

**Comment by alanmcruickshank** [comment]
If nothing else we should increase test coverage for the `{#- ... #}` syntax and equivalents. For example it probably won't work with the `L046` rule I just put together for linting whitespace within jinja tags.

That might also highlight more issues with the parsing code too. Thanks for the bug report!

**Comment by barrywhart** [comment]
I'll look at this issue if I have time. I just fixed a somewhat similar issue, #1321. As part of this work, it would be cool to enhance SQLFluff's core template processing to detect if the source positions are inaccurate wrt the actual input file. That could provide early detection of bugs, rather than waiting until this results in `sqlfluff fix` corrupting a file.
